includes/class-wc-cart.php

```diff
@@ -1056,7 +1055,7 @@ public function add_to_cart( $product_id = 0, $quantity = 1, $variation_id = 0,
 						}
 
 						// Don't include if it's empty.
-						if ( ! empty( $value ) ) {
+						if ( ! empty( $value ) || '0' === $value ) {
```

Comment:
why not check for `in_array( $value, $attribute['options'] )` here?

Reply:
My thinking was that we should try to keep this as minimal a change as possible to avoid potential backwards compatibility breaks. The original behavior was to add every value to the array prior to the fix in #27115 and given that most of this is filterable I feel this is correct enough.

Reply:
ok, make sense to have as minimum changes as possible. resolving the conversation.